### PR TITLE
Add scripts for schema auto-generation

### PR DESCRIPTION
**Pull request by antmarti-microsoft**

### CODE REVIEW

File: schemas/2019-04-01/deploymentTemplate.json [hunk 1]
```diff
@@ -1100,6 +1100,9 @@
                 ]
               }
             ]
+          },
+          { 
+            "$ref": "https://schema.management.azure.com/schemas/2019-04-01/generatedResources.json"
```
Comment (TianoMS) [inline schemas/2019-04-01/deploymentTemplate.json]: >"$ref": "https://schema.management.azure.com/schemas/2019-04-01/generatedResources.json" [](start = 12, length = 88)

looks like this causes a test failure.

File: README.md [hunk 2]
```diff
@@ -4,24 +4,39 @@ This is the repo for template deployment schemas hosted at https://schema.manage
...
+You can use the generator in this repo to automatically generate a schema from a swagger spec checked into the [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs) repo. We are working on fully automating this process, but please note that until then, it is your responsibility to ensure that the auto-generated schema has been correctly formatted before submitting a pull request.
+### Instructions
+1. Fork this repo, and clone it locally.
+2. Run the following commands (replace the provider namespace and api version accordingly):
+    * `cd generator`
+    * `npm install`
+    * `npm run generate-single -- -ProviderNamespace Microsoft.MyProvider -ApiVersion 2019-04-01`
+4. Review the generator logs to ensure no errors, and review the changes generated.
+5. Generate a commit and push it to your fork.
+6. Submit a pull request to this repo.
```
Comment (TianoMS) [inline README.md]: this is cool. do we want partners to start using it? Are we going to stay in this stage for long time?

Comment (antmarti-microsoft) [reply README.md]: I don't see why not. It may be useful for helping us identify bugs.

File: schemas/2019-04-01/deploymentTemplate.json (hunk 1 above)
Comment (antmarti-microsoft) [reply schemas/2019-04-01/deploymentTemplate.json]: Seems like it was unhappy with an empty "oneOf" statement. I've used `npm run generate-all` to add the previously auto-generated resources to this file.

**Comment by Tiano2017** [comment]
seems there are conflicts.

**Pull request merged**